### PR TITLE
Py 3.4 support was removed: Upgrade lxml

### LINKED ISSUE #49: Conflicting versions of lxml
* ricecooker version: 0.6.61
* Python version: 3.6

### Description

So pip is not thaaaat great at dependencies, but at least now it's good at finding out about conflicts. In this case, `lxml` is resolved by the top-level dependency pressurecooker, but later contradicted by a transient dependency.

Can be resolved by:

* adding `le-pycaption`'s version target in pressurecooker
* updating le-pycaption with a higher lxml max version (which is great)
* relaxing the lxml max version in le-pycaption

I'll try out ricecooker with lxml 4.5.0 and see how that goes.

Cutting out the various parts of below output, here's the issue:

```
  - pressurecooker [required: >=0.0.27, installed: 0.0.27]
    - EbookLib [required: >=0.17.1, installed: 0.17.1]
      - lxml [required: Any, installed: 4.5.0]
    - le-pycaption [required: >=2.1.0a2, installed: 2.1.0a2]
      - lxml [required: >=3.2.3,<4.4.0, installed: 4.5.0]
```

Installing latest ricecooker:

```
ERROR: le-pycaption 2.1.0a2 has requirement lxml<4.4.0,>=3.2.3, but you'll have lxml 4.5.0 which is incompatible.
```

Output from pipdeptree:

```
➜  sushi-chef-ubongokids git:(2020-update) ✗ pipdeptree                   
Warning!!! Possibly conflicting dependencies found:
* le-pycaption==2.1.0a2
 - lxml [required: >=3.2.3,<4.4.0, installed: 4.5.0]
------------------------------------------------------------------------
pipdeptree==0.13.2
  - pip [required: >=6.0.0, installed: 20.0.2]
ricecooker==0.6.42
  - beautifulsoup4 [required: >=4.6.3, installed: 4.9.0]
    - soupsieve [required: >1.2, installed: 2.0]
  - cachecontrol [required: ==0.12.0, installed: 0.12.0]
    - msgpack-python [required: Any, installed: 0.5.6]
    - requests [required: Any, installed: 2.23.0]
      - certifi [required: >=2017.4.17, installed: 2020.4.5.1]
      - chardet [required: >=3.0.2,<4, installed: 3.0.4]
      - idna [required: >=2.5,<3, installed: 2.9]
      - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.9]
  - colorlog [required: >=4.1.0,<4.2, installed: 4.1.0]
  - css-html-js-minify [required: ==2.2.2, installed: 2.2.2]
    - anglerfish [required: Any, installed: 2.5.0]
  - dictdiffer [required: >=0.8.0, installed: 0.8.1]
  - docopt [required: >=0.6.2, installed: 0.6.2]
  - html5lib [required: Any, installed: 1.0.1]
    - six [required: >=1.9, installed: 1.14.0]
    - webencodings [required: Any, installed: 0.5.1]
  - le-utils [required: >=0.1.24, installed: 0.1.24]
    - pycountry [required: ==17.5.14, installed: 17.5.14]
  - lockfile [required: ==0.12.2, installed: 0.12.2]
  - mock [required: ==2.0.0, installed: 2.0.0]
    - pbr [required: >=0.11, installed: 5.4.5]
    - six [required: >=1.9, installed: 1.14.0]
  - Pillow [required: ==5.4.1, installed: 5.4.1]
  - pressurecooker [required: >=0.0.27, installed: 0.0.27]
    - beautifulsoup4 [required: >=4.6.3, installed: 4.9.0]
      - soupsieve [required: >1.2, installed: 2.0]
    - EbookLib [required: >=0.17.1, installed: 0.17.1]
      - lxml [required: Any, installed: 4.5.0]
      - six [required: Any, installed: 1.14.0]
    - ffmpy [required: >=0.2.2, installed: 0.2.2]
    - le-pycaption [required: >=2.1.0a2, installed: 2.1.0a2]
      - beautifulsoup4 [required: >=4.6.3, installed: 4.9.0]
        - soupsieve [required: >1.2, installed: 2.0]
      - cssutils [required: >=0.9.10, installed: 1.0.2]
      - future [required: Any, installed: 0.18.2]
      - lxml [required: >=3.2.3,<4.4.0, installed: 4.5.0]
      - six [required: >=1.9.0, installed: 1.14.0]
    - le-utils [required: >=0.1.24, installed: 0.1.24]
      - pycountry [required: ==17.5.14, installed: 17.5.14]
    - matplotlib [required: ==2.2.3, installed: 2.2.3]
      - cycler [required: >=0.10, installed: 0.10.0]
        - six [required: Any, installed: 1.14.0]
      - kiwisolver [required: >=1.0.1, installed: 1.2.0]
      - numpy [required: >=1.7.1, installed: 1.15.4]
      - pyparsing [required: >=2.0.1,!=2.1.6,!=2.1.2,!=2.0.4, installed: 2.4.7]
      - python-dateutil [required: >=2.1, installed: 2.8.1]
        - six [required: >=1.5, installed: 1.14.0]
      - pytz [required: Any, installed: 2019.3]
      - six [required: >=1.10, installed: 1.14.0]
    - numpy [required: ==1.15.4, installed: 1.15.4]
    - pdf2image [required: ==1.11.0, installed: 1.11.0]
      - pillow [required: Any, installed: 5.4.1]
    - Pillow [required: ==5.4.1, installed: 5.4.1]
    - youtube-dl [required: >=2020.1.24, installed: 2020.3.24]
  - pypdf2 [required: >=1.26.0, installed: 1.26.0]
  - pytest [required: >=3.0.2, installed: 5.4.1]
    - attrs [required: >=17.4.0, installed: 19.3.0]
    - importlib-metadata [required: >=0.12, installed: 1.6.0]
      - zipp [required: >=0.5, installed: 3.1.0]
    - more-itertools [required: >=4.0.0, installed: 8.2.0]
    - packaging [required: Any, installed: 20.3]
      - pyparsing [required: >=2.0.2, installed: 2.4.7]
      - six [required: Any, installed: 1.14.0]
    - pluggy [required: >=0.12,<1.0, installed: 0.13.1]
      - importlib-metadata [required: >=0.12, installed: 1.6.0]
        - zipp [required: >=0.5, installed: 3.1.0]
    - py [required: >=1.5.0, installed: 1.8.1]
    - wcwidth [required: Any, installed: 0.1.9]
  - requests [required: >=2.11.1, installed: 2.23.0]
    - certifi [required: >=2017.4.17, installed: 2020.4.5.1]
    - chardet [required: >=3.0.2,<4, installed: 3.0.4]
    - idna [required: >=2.5,<3, installed: 2.9]
    - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.9]
  - requests-cache [required: >=0.4.13, installed: 0.5.2]
    - requests [required: >=1.1.0, installed: 2.23.0]
      - certifi [required: >=2017.4.17, installed: 2020.4.5.1]
      - chardet [required: >=3.0.2,<4, installed: 3.0.4]
      - idna [required: >=2.5,<3, installed: 2.9]
      - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.9]
  - requests-file [required: Any, installed: 1.4.3]
    - requests [required: >=1.0.0, installed: 2.23.0]
      - certifi [required: >=2017.4.17, installed: 2020.4.5.1]
      - chardet [required: >=3.0.2,<4, installed: 3.0.4]
      - idna [required: >=2.5,<3, installed: 2.9]
      - urllib3 [required: >=1.21.1,<1.26,!=1.25.1,!=1.25.0, installed: 1.25.9]
    - six [required: Any, installed: 1.14.0]
  - selenium [required: ==3.0.1, installed: 3.0.1]
  - validators [required: Any, installed: 0.14.3]
    - decorator [required: >=3.4.0, installed: 4.4.2]
    - six [required: >=1.4.0, installed: 1.14.0]
  - websocket-client [required: ==0.40.0, installed: 0.40.0]
    - six [required: Any, installed: 1.14.0]
  - youtube-dl [required: Any, installed: 2020.3.24]
setuptools==46.1.3
shelve2==1.0
wheel==0.34.2
```

### PR DESCRIPTION
**Pull request by benjaoming**
Bump up the lower bound, remove upper bound.

Fixes: https://github.com/learningequality/pressurecooker/issues/49

This is a bit fly-by, but I understood that the project no longer needs to support Python 2.7 and 3.4 which was the reason for this version restriction.

It's really really confusing that the working branch is called `py27`. Can we fix that, too?

### CODE REVIEW

**Comment by bjester** [comment]
It looks like the tests are failing but it also looks like they're failing on the `py27` branch too. I suspect because I accidentally did not recreate the tox env when I ran the tests after previous changes :frowning_face: 

The `py27` originates from the PBS package. This is based off a `py27-py3-compat` branch that was based off `py27`, so when they eventually merged the compat branch, I did the same. Most of the important updates are still in PR (limbo) against the PBS package.

File: setup.py [hunk 1]
```diff
@@ -9,7 +9,7 @@
 
 dependencies = [
     'beautifulsoup4>=4.6.3',
-    'lxml>=3.2.3,<4.4.0',
+    'lxml>=4.5.0',
```
Comment (ivanistheone) [inline setup.py:12]: Looking at the other forks pycaption forks, I this requirement is relaxed simply as `lxml>=3.2.3` --- no need to push for the latest. Let other packages figure which exact version needed.

**Comment by benjaoming** [comment]
> Most of the important updates are still in PR (limbo) against the PBS package.

@bjester true, but there are some nice forks, see: https://github.com/learningequality/pressurecooker/issues/50

**Comment by bjester** [comment]
The test issues were from `beautifulsoup4==4.9.0`. @ivanistheone is adding back Python 2.7 with an enum compat lib, and will include this as well, so we can close.

**Pull request closed without merging**